src/main/java/io/fusionauth/http/server/HTTPRequest.java

```diff
@@ -738,6 +738,9 @@ private void decodeHeader(String name, String value) {
         int colon = value.indexOf(':');
         if (colon > 0) {
           this.host = value.substring(0, colon);
+          String portString = value.substring(colon + 1);
+          if (portString.trim().length() > 0)
```

Comment:
Is this necessary? I don't think you can have spaces around port numbers per spec.

Also, we use the java coding standards so this needs curly braces for the if-statement.

Finally, this could fail really hard if the `parseInt` fails. This should protect against bad values and fall back to the server port.